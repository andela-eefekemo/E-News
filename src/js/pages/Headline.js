import React from 'react';

import Nav from '../components/layout/Nav';
import Footer from '../components/layout/Footer';

export default class Headlines extends React.Component {
  render() {
    console.log(this.props);
    return (
      <div>
        <Nav />
        <h1>the headlines</h1>
        <Footer />
      </div>
    );
  }
}
