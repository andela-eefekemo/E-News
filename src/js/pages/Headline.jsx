import React from 'react';

import PropTypes from 'prop-types';
import Nav from '../components/layout/Nav.jsx';
import Footer from '../components/layout/Footer.jsx';
import NewsStore from '../stores/newsStore';
import * as NewsActions from '../actions/newsAction';
import Dropdown from '../components/Dropdown.jsx';
import Articles from '../components/Articles.jsx';

/**
 * Headlines React Component
 */
class Headlines extends React.Component {
  constructor() {
    super();
    this.api = NewsStore.getArticles();
    this.state = {
      articles: this.api,
      name: 'articles',
    };
    this.article = this.getArticle.bind(this);
    this.sortsBy = [];
    this.name = '';
    this.id = '';
    this.change = this.onChange.bind(this);
  }
  // componentWillMount - Runs when component is loaded
  componentWillMount() {
    const { location } = this.props;
    const { query } = location;
    const sorts = query.sorts.split(',');
    this.sortsBy = sorts.map(sort => sort);
    this.name = query.name;
    this.id = query.id;
    NewsStore.on('changes', this.article);
    NewsActions.getArticle(query.id, sorts[0]);
  }
  // componentWillUnMount - Runs when component is unloaded(changed)
  componentWillUnmount() {
    NewsStore.removeListener('changes', this.article);
  }
  // onChange - Runs when onChange event is called in select field
  onChange(event) {
    const value = event.target.value;
    NewsActions.getArticle(this.id, value);
  }
  // getArticle - Sets state of article
  getArticle() {
    this.setState({
      articles: NewsStore.getArticles(),
    });
  }
  render() {
    // maps through sortsBy array and passes props to dropdown component
    const sorted = this.sortsBy
      .map((sort, index) => <Dropdown key={index} value={sort} text={sort} />);
    // maps through articles array and passes props to Articles component
    const singleArticles = this.state.articles
      .map(article => <Articles key={article.publishedAt} {...article} />);
    return (
      <div>
        <Nav />
        <div className="container" id={this.state.name}>
          <div className="row">
            <div className="col m8">
              <h1>{this.name}</h1>
            </div>
            <div className="col m4">
              <select
                className="browser-default input-field"
                onChange={this.change} >
                {sorted}
              </select>
            </div>
          </div>
          <div className="row">
            {singleArticles}
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
// sets default propstype
Headlines.defaultProps = {
  location: {},
};

Headlines.propTypes = {
  location: PropTypes.object,
};

export default Headlines;
