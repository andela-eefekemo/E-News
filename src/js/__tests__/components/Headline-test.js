import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Headline from '../../pages/Headline';


test('Headline Component', () => {
  const component = shallow(<Headline />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
