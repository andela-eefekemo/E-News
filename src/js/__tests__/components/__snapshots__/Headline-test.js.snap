// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render a button 1`] = `
<div
  className="col s12 m6 l6 height"
>
  <div
    className="card card-panel hoverable"
  >
    <div
      className="card-image"
    >
      <img
        alt="Headline"
        src=""
      />
      <span
        className="card-title"
      >
        
      </span>
    </div>
    <div
      className="card-content"
    >
      <p
        className="truncate"
      >
        
      </p>
    </div>
    <div
      className="card-action"
    >
      <a
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        View More
      </a>
    </div>
  </div>
</div>
`;

exports[`Headline Component 1`] = `
<Headlines
  location={
    Object {
      "query": Object {
        "sorts": "top",
      },
    }
  }
  sorts="top"
>
  <div>
    <Nav
      imageURL=""
      info={
        Object {
          "imageURL": "",
          "name": "",
        }
      }
      name=""
    >
      <div>
        <ul
          className="dropdown-content"
          id="dropdown1"
        >
          <li>
            <img
              alt="profile"
              className="pic"
              src=""
            />
          </li>
          <li
            className="divider"
          />
          <li>
            <Link
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/logout"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Logout
              </a>
            </Link>
          </li>
        </ul>
        <nav>
          <div
            className="nav-wrapper teal lighten-1"
          >
            <a
              className="brand-logo"
              href="#!"
            >
              E-News
            </a>
            <a
              className="button-collapse"
              data-activates="mobile-demo"
              href="#!"
            >
              <img
                alt="profile"
                className="pic"
                src=""
              />
            </a>
            <ul
              className="right hide-on-med-and-down"
            >
              <li>
                <IndexLink
                  to="/"
                >
                  <Link
                    onlyActiveOnIndex={true}
                    style={Object {}}
                    to="/"
                  >
                    <a
                      onClick={[Function]}
                      style={Object {}}
                    >
                      Home
                    </a>
                  </Link>
                </IndexLink>
              </li>
              <li>
                <a
                  className="dropdown-button"
                  data-activates="dropdown1"
                  href="#!"
                >
                  <i
                    className="material-icons right"
                  >
                    arrow_drop_down
                  </i>
                </a>
              </li>
            </ul>
            <ul
              className="side-nav"
              id="mobile-demo"
            >
              <li>
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/"
                >
                  <a
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="divider"
              />
              <li>
                 
                <Link
                  onClick={[Function]}
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/logout"
                >
                  <a
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Logout
                  </a>
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </Nav>
    <div
      className="container"
      id="articles"
    >
      <div
        className="row"
      >
        <div
          className="col m8"
        >
          <h1 />
        </div>
        <div
          className="col m4"
        >
          <select
            className="browser-default input-field"
            onChange={[Function]}
          />
        </div>
      </div>
      <div
        className="row"
      />
    </div>
    <Footer>
      <div
        className="footer"
      >
        <p
          className="center-align"
        >
          Â© 2017 | E-News
        </p>
        <p
          className="center-align"
        >
          Developed by: Efekemo Eguonoghene
        </p>
      </div>
    </Footer>
  </div>
</Headlines>
`;
